### PR TITLE
Adds Pybind as a submodule and uses v2.6

### PR DESCRIPTION
**Pull request by RawnH**
This PR adds PyBind11 as a submodule to TACO and removes the original folder. The submodule will be automatically pulled by cmake when building the python bindings.

~~This is in draft since it look like one of the tests break with this update. I will remove from draft once this is fixed.~~

This attempts to address issue #397 and make it easier to switch versions in the future.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -108,9 +108,29 @@ add_subdirectory(test)
 add_subdirectory(tools)
 add_subdirectory(apps)
 string(REPLACE " -Wmissing-declarations" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+
 if(PYTHON)
+  find_package(Git QUIET)
+  if(GIT_FOUND AND EXISTS "${TACO_PROJECT_DIR}/.git")
+    # Update submodules as needed
+    option(GIT_SUBMODULE "Check submodules during build" ON)
```
Comment (Infinoid) [inline CMakeLists.txt:116]: This is great.  But could we make this submodule checkout stuff globally available?  I'd like to use it for googletest and bats (see #393) too, and I'd like to be able to clone those submodules even when PYTHON is disabled.

**Review by Infinoid — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (RawnH) [reply CMakeLists.txt:116]: Sounds good. I will just pull pybind unconditionally then.

Comment (Infinoid) [reply CMakeLists.txt]: It looks good to me, thanks!

I don't see a way to mark it as resolved... maybe I am blind?

**Comment by stephenchouca** [comment]
I'll go ahead and merge this into master then.

**Pull request merged**